Automattic/woocommerce-payments
# Add document download link and handler

Fixes #3942 

#### Changes proposed in this Pull Request

This PR adds a new API endpoint that will request a document from the server and serve it to the merchant's browser.

It also adds a download link for each document on the documents list. Clicking this link will open the document on a new tab or window.

#### Testing instructions

* Navigate to the Documents page
* Make sure that each document has a "Download" link
* Click on the download link for a document and make sure it is opened in a new tab
* Modify the URL of the opened file to an invalid `document_id` or `_wpnonce` and make sure the request fails
* Test a few of the WC Pay screens to make sure the refactor of the API Client `request` method didn't break anything

-------------------


**Post merge**



- [x] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) : _'QA Testing Not Applicable'_
